prow/scripts/cluster-integration/reconciler-gardener-long-lasting.sh

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env bash
+
+set -e
+set -o pipefail
+
+#Description: Kyma CLI Integration plan on Gardener. This scripts implements a pipeline that consists of many steps. The purpose is to install and test Kyma using the CLI on a real Gardener cluster.
+#
+#Expected common vars:
+# - JOB_TYPE - set up by prow (presubmit, postsubmit, periodic)
+# - KYMA_PROJECT_DIR - directory path with Kyma sources to use for installation
+# - GARDENER_REGION - Gardener compute region
+# - GARDENER_ZONES - Gardener compute zones inside the region
+# - GARDENER_CLUSTER_VERSION - Version of the Kubernetes cluster
+# - GARDENER_KYMA_PROW_KUBECONFIG - Kubeconfig of the Gardener service account
+# - GARDENER_KYMA_PROW_PROJECT_NAME - Name of the gardener project where the cluster will be integrated.
+# - GARDENER_KYMA_PROW_PROVIDER_SECRET_NAME - Name of the secret configured in the gardener project to access the cloud provider
+# - MACHINE_TYPE - (optional) machine type
+#
+#Please look in each provider script for provider specific requirements
+
+function delete_cluster_if_exists(){
```

Comment:
Did you check if you can use functions from lib directory? **https://github.com/kyma-project/test-infra/tree/main/prow/scripts/lib/gardener**

We have functions for creating, removing and connecting to shoot clusters.

Reply:
The creation of shoot clusters in the library is different from my need. Hence I’d stick to my approach. The deletion can be reused but with a check, if exists. I reused `gardener::deprovision_cluster` for now. Connecting to shoot is missing completely.